src/cpp/rtps/messages/RTPSMessageGroup.cpp

```diff
@@ -381,12 +381,21 @@ bool RTPSMessageGroup::add_info_ts_in_buffer(const std::vector<GUID_t>& remote_r
     uint32_t from_buffer_position = submessage_msg_->pos;
 #endif
 
-    // Insert INFO_TS submessage.
-    // TODO (Ricardo) Source timestamp maybe has marked when user call write function.
-    if(!RTPSMessageCreator::addSubmessageInfoTS_Now(submessage_msg_, false)) //Change here to add a INFO_TS for DATA.
+    if (timestamp == Time_t())
     {
```

Comment:
I don't know but because the code was using `addSubmessageInfoTS_Now` I thought I would leave it just in case. Should I just remove this `if` and only do `RTPSMessageCreator::addSubmessageInfoTS(submessage_msg_, timestamp, false)´?